dsl_parser/elements/node_templates.py

```diff
@@ -149,7 +149,7 @@ def parse(self, relationships, data_types):
 class NodeTemplateInstancesDeploy(Element):
 
     required = True
-    schema = Leaf(type=int)
+    schema = Leaf(type=(int, dict))
 
     def validate(self):
         if self.initial_value < 0:
```

Comment:
(1) is indeed not the way to go
i think i might have opted for (2) in this case, only in this specific code. i agree with you that in general we'd want the intrinsic functions framework to support better validation, but right now it's a change we'd rather avoid because of other refactorings that are happening in the DSL parser.
either way, i believe some of the code in this class should be modified, e.g. in the validate function wont the code fail over the initial_value if the value passed is a dict?